vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java

```diff
@@ -29,17 +54,86 @@ public void setExhibition(JoinPoint jp) {
-                    if (!((Model) obj).containsAttribute("exhibition")) {
+        if(args!=null && returnType==String.class) {
+            for(Object obj : args) {
+                if(obj instanceof Model && !(obj instanceof RedirectAttributes)) {
+                    if(!((Model) obj).containsAttribute("exhibition")) {
                         ((Model) obj).addAttribute("exhibition", exhibitionManager.getStartExhibition());
                     }
-                    if (!((Model) obj).containsAttribute("allSpaces")) {
+                    if(!((Model) obj).containsAttribute("allSpaces")) {
                         ((Model) obj).addAttribute("allSpaces", spaceManager.getAllSpaces());
                     }
                 }
+            }  
+        }
+    }
+
+    @Around("execution(public * edu.asu.diging.vspace.web.publicview..*Controller.*(..))")
+    public Object showExhibition(ProceedingJoinPoint jp) throws Throwable {
+        Object[] args = jp.getArgs();
+        MethodSignature signature = (MethodSignature) jp.getSignature();
+        int indexofModel = (Arrays.asList(signature.getParameterTypes())).indexOf(Model.class);
+        Exhibition exhibition = (Exhibition) exhibitionManager.getStartExhibition(); 
+        if(exhibition==null) {
+            return "home";
+        }
+        if(exhibition.getMode() == null) {
+            return jp.proceed();
+        }
+        Map<String, String> ids = getIds(args, signature);
+        String spaceId = ids.getOrDefault("spaceId", "");
+        String moduleId = ids.getOrDefault("moduleId", "");
+        if(spaceId.equals("") && moduleId.equals("") && 
+            (exhibition.getMode().equals(ExhibitionModes.ACTIVE) || authFacade.getAuthenticatedUser()!=null)) {
+            return "notFound";
+        }
+        return redirectRequest(jp, spaceId, moduleId, indexofModel, exhibition);
+    }
+    
+    public Object redirectRequest(ProceedingJoinPoint jp, String spaceId, String moduleId, int modelIndex, Exhibition exhibition) throws Throwable{
+        ISpace space = spaceManager.getSpace(spaceId);
+        IModule module = moduleManager.getModule(moduleId);
+        Object[] args = jp.getArgs();
+        if(exhibition.getMode().equals(ExhibitionModes.ACTIVE)) {
+            if(space==null && module==null) {
+                return "notFound";
+            }
+            else {
+                return jp.proceed();
+            }
+        }
+        if(authFacade.getAuthenticatedUser()!=null && modelIndex>-1) {
+            if(space==null && module==null) {
+                return "notFound";
+            }
+            else {
+                ((Model) args[modelIndex]).addAttribute("showModal", "true");
+                return jp.proceed();
+            }
+        }
+        if(exhibition.getMode()==ExhibitionModes.OFFLINE && !exhibition.getCustomMessage().equals("")) {
+            ((Model) args[modelIndex]).addAttribute("modeValue", exhibition.getCustomMessage());
+        } else {
```

Comment:
so this else block will always be executed when either the mode is not offline or there is no custom message? that seems wrong. also, it would be much cleaner to have a tertiary expression set the custom message and then add the attribute once.

Reply:
The else block will only handle maintenance as Active is handled before and Offline is handled in if. I will implement tertiary statement for offline.